src/compiler/transformers/module/module.ts

```diff
@@ -3,16 +3,756 @@
+                if (emitAsReturn) {
+                    return createReturn(exportEquals.expression);
+                }
+                else {
+                    return createStatement(
+                        createAssignment(
+                            createPropertyAccess(
+                                createIdentifier("module"),
+                                "exports"
+                            ),
+                            exportEquals.expression
+                        )
+                    );
+                }
+            }
+            return undefined;
+        }
+
+        function getModuleMemberName(name: Identifier) {
+            return createPropertyAccess(
+                createIdentifier("exports"),
+                name,
+                /*location*/ name
+            );
+        }
+
+        function getExternalModuleNameLiteral(importNode: ImportDeclaration | ExportDeclaration | ImportEqualsDeclaration) {
+            const moduleName = getExternalModuleName(importNode);
+            if (moduleName.kind === SyntaxKind.StringLiteral) {
+                return tryRenameExternalModule(<StringLiteral>moduleName)
+                    || getSynthesizedClone(<StringLiteral>moduleName);
+            }
+
+            return undefined;
+        }
+
+        /**
+         * Some bundlers (SystemJS builder) sometimes want to rename dependencies.
+         * Here we check if alternative name was provided for a given moduleName and return it if possible.
+         */
+        function tryRenameExternalModule(moduleName: LiteralExpression) {
+            if (currentSourceFile.renamedDependencies && hasProperty(currentSourceFile.renamedDependencies, moduleName.text)) {
+                return createLiteral(currentSourceFile.renamedDependencies[moduleName.text]);
+            }
+            return undefined;
+        }
+
+        function getLocalNameForExternalImport(node: ImportDeclaration | ExportDeclaration | ImportEqualsDeclaration): Identifier {
+            const namespaceDeclaration = getNamespaceDeclarationNode(node);
+            if (namespaceDeclaration && !isDefaultImport(node)) {
+                return createIdentifier(getSourceTextOfNodeFromSourceFile(currentSourceFile, namespaceDeclaration.name));
+            }
+            if (node.kind === SyntaxKind.ImportDeclaration && (<ImportDeclaration>node).importClause) {
+                return getGeneratedNameForNode(node);
+            }
+            if (node.kind === SyntaxKind.ExportDeclaration && (<ExportDeclaration>node).moduleSpecifier) {
+                return getGeneratedNameForNode(node);
+            }
+        }
+
+        function createRequireCall(importNode: ImportDeclaration | ImportEqualsDeclaration | ExportDeclaration) {
+            const moduleName = getExternalModuleNameLiteral(importNode);
+            return createCall(
+                createIdentifier("require"),
+                moduleName ? [moduleName] : []

```

Comment:
without `moduleName`, this just generates `require()`, right? What will the larger generated statement be in that case, and is it meaningful?